src/Elastic.Transport/Responses/DefaultResponseFactory.cs

```diff
@@ -122,17 +124,21 @@ private async ValueTask<TResponse> CreateCoreAsync<TResponse>(
 		response ??= new TResponse();
 		response.ApiCallDetails = details;
 		return response;
-	}	
+	}
 
-	private	bool TryResolveBuilder<TResponse>(out IResponseBuilder builder) where TResponse : TransportResponse, new()
+	private	bool TryResolveBuilder<TResponse>(
+		out IResponseBuilder builder,
+		IReadOnlyCollection<IResponseBuilder> productResponseBuilders,
+		IReadOnlyCollection<IResponseBuilder> responseBuilders
+	) where TResponse : TransportResponse, new()
```

Comment:
nit: maybe reorder so `responseBuilders` is first as they are registered first

Reply:
Good call, `out` parameters should always be last too.